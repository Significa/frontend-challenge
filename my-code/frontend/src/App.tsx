import React from "react";
import GlobalStyles from "./styles/global";
import { ThemeProvider } from "styled-components";
import { BrowserRouter } from "react-router-dom";
import dark from "./themes/dark";
import Routes from "./routes";
import Header from "./components/Header";
import LimitWidth from "./components/LimitWidth";

const App = () => {
  return (
    <>
      <ThemeProvider theme={dark}>
        <GlobalStyles />
        <LimitWidth>
          <Header />
          <BrowserRouter>
            <Routes />
          </BrowserRouter>
        </LimitWidth>
      </ThemeProvider>
    </>
  );
};

export default App;
