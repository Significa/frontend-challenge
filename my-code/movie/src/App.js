import React, { Component } from 'react'
import { BrowserRouter as Router, Route } from 'react-router-dom'

import Home from './routes/home'
import Details from './routes/details'

import { movieSearch, movieDetails } from './api'

const movies = {"Search":[{"Title":"Star Wars: Episode IV - A New Hope","Year":"1977","imdbID":"tt0076759","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BNzVlY2MwMjktM2E4OS00Y2Y3LWE3ZjctYzhkZGM3YzA1ZWM2XkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg"},{"Title":"Star Wars: Episode V - The Empire Strikes Back","Year":"1980","imdbID":"tt0080684","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BYmU1NDRjNDgtMzhiMi00NjZmLTg5NGItZDNiZjU5NTU4OTE0XkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg"},{"Title":"Star Wars: Episode VI - Return of the Jedi","Year":"1983","imdbID":"tt0086190","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BOWZlMjFiYzgtMTUzNC00Y2IzLTk1NTMtZmNhMTczNTk0ODk1XkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg"},{"Title":"Star Wars: The Force Awakens","Year":"2015","imdbID":"tt2488496","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BOTAzODEzNDAzMl5BMl5BanBnXkFtZTgwMDU1MTgzNzE@._V1_SX300.jpg"},{"Title":"Star Wars: Episode I - The Phantom Menace","Year":"1999","imdbID":"tt0120915","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BYTRhNjcwNWQtMGJmMi00NmQyLWE2YzItODVmMTdjNWI0ZDA2XkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg"},{"Title":"Star Wars: Episode III - Revenge of the Sith","Year":"2005","imdbID":"tt0121766","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BNTc4MTc3NTQ5OF5BMl5BanBnXkFtZTcwOTg0NjI4NA@@._V1_SX300.jpg"},{"Title":"Star Wars: Episode II - Attack of the Clones","Year":"2002","imdbID":"tt0121765","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BMDAzM2M0Y2UtZjRmZi00MzVlLTg4MjEtOTE3NzU5ZDVlMTU5XkEyXkFqcGdeQXVyNDUyOTg3Njg@._V1_SX300.jpg"},{"Title":"Star Trek","Year":"2009","imdbID":"tt0796366","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BMjE5NDQ5OTE4Ml5BMl5BanBnXkFtZTcwOTE3NDIzMw@@._V1_SX300.jpg"},{"Title":"Rogue One: A Star Wars Story","Year":"2016","imdbID":"tt3748528","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BMjEwMzMxODIzOV5BMl5BanBnXkFtZTgwNzg3OTAzMDI@._V1_SX300.jpg"},{"Title":"Star Trek: Into Darkness","Year":"2013","imdbID":"tt1408101","Type":"movie","Poster":"https://m.media-amazon.com/images/M/MV5BMTk2NzczOTgxNF5BMl5BanBnXkFtZTcwODQ5ODczOQ@@._V1_SX300.jpg"}],"totalResults":"3086","Response":"True"}

class App extends Component {

  constructor(props) {
    super(props)
    this.state = {
      search: "",
      detail: "",
      fav: [],
      results: movies.Search//false
    }

    this.changeText = this.changeText.bind(this)
    this.getMovies = this.getMovies.bind(this)
    this.getDetails = this.getDetails.bind(this)
    this.handleLikes = this.handleLikes.bind(this)
  }

  handleLikes(id) {    
    const remove = (filmID) => {this.state.fav.filter(fav => {
      return fav !== filmID
    })}

    if(this.state.fav.includes(id)){
      let newFav = remove(id)
      return this.setState({fav: newFav})
    }
    this.setState({fav: [id, ...this.state.fav]})
  }

  changeText(event) {
    this.setState({
      search: event.target.value
    })
  }

  async getDetails(id) {
    let detail = await movieDetails(id)
    this.setState({detail})
    console.log(detail)
  }

  async getMovies() {    
    let search = await movieSearch(this.state.search)
    this.setState({results: search.Search})  
  }

  render() {
    return (
      <Router>
        <div className="App">        
          <Route exact path = '/' render={(props) => <Home {...props} res={this.state.results} search={this.state.value} searchHandler={this.changeText} submit={this.getMovies}/>} />
          <Route path = '/details/:id' render={(props) => <Details {...props} fetch={this.getDetails} detail={this.state.detail} like={this.handleLikes}/>} />
        </div>
        
        {/* <div className="App">
          <header className="App-header">
            <h1 className="App-title">Welcome to React</h1>
          </header>
          <p className="App-intro">
            To get started, edit <code>src/App.js</code> and save to reload.
          </p>
          <input type="text" value={this.state.search} onChange={this.changeText}/>
          <button onClick={this.getMovies}>Click</button>
          {this.state.results && 
            <ul>
              {this.state.results.map(cur => {
                return (<li key={cur.imdbID} onClick={() => this.getDetails(cur.imdbID)}>{cur.Title}</li>)
              })}
            </ul>
          }
        </div> */}
      </Router>
      
    );
  }
}

export default App
