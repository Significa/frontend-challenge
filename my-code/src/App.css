* {
  font-family: Roboto, sans-serif;
}

.App {

  background-color: #282c34;
  text-align: center;
  height: 96vh;
}
