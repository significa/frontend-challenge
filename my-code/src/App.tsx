import { BrowserRouter } from "react-router-dom"
import { MoviesDataProvider } from "./Contexts/MoviesData"
import { Router } from "./Pages/Router"

function App() {

  return (
    <BrowserRouter>
      <MoviesDataProvider>
        <Router />
      </MoviesDataProvider>
    </BrowserRouter>
  )
}

export default App
