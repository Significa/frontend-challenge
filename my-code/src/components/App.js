import React from "react";
import {
    BrowserRouter as Router,
    Switch,
    Route,
} from "react-router-dom";
import { FetchMovie } from "./FetchMovie";
import { Movies } from "./Movies";

export function App() {
    return (
        <Router>
            <div>
                <nav class="navbar navbar-expand-lg navbar-light bg-light">
                    <a class="navbar-brand" href="/">MOVIE SEARCH!</a>
                </nav>
                <Switch>
                    <Route exact path="/">
                        <Movies />
                    </Route>
                    <Route path="/title/:id">
                        <FetchMovie />
                    </Route>

                </Switch>
            </div>
        </Router>
    );
}