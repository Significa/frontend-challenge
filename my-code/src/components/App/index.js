import React from 'react'
import PropTypes from 'prop-types'
import { Route, Switch } from 'react-router'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'connected-react-router'
import { ThemeProvider } from 'styled-components'

import Layout from '../Layout'
import Search from '../Search'
import Movie from '../../containers/Movie'

export default function App({ history, store, theme }) {
  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <ThemeProvider theme={theme}>
          <Layout>
            <Switch>
              <Route path="/movie" component={Movie} />
              <Route component={Search} />
            </Switch>
          </Layout>
        </ThemeProvider>
      </ConnectedRouter>
    </Provider>
  )
}

App.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  history: PropTypes.object.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  store: PropTypes.object.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  theme: PropTypes.object.isRequired
}
