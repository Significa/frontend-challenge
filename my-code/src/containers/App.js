// @flow
import React from "react"
import { BrowserRouter, Route, Switch } from "react-router-dom"

import Main from "./Main"
import Detail from "./Detail"
import TopBar from "../layout/topbar"

import { Flex } from "../layout/Layout"

const App = () => (
  <BrowserRouter>
    <Flex>
      <Route path="/" component={TopBar} />
      <Switch>
        <Route exact path="/" component={Main} />
        <Route exact path="/movie/:id" component={Detail} />
      </Switch>
    </Flex>
  </BrowserRouter>
)

export default App
