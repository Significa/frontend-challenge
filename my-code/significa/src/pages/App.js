import React from "react";
import Layout from "../components/layout";
import "./App.css";

function App() {
  return <Layout>working... yeahh!!!</Layout>;
}

export default App;
