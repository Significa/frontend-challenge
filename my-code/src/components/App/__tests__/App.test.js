import React from 'react'
import { shallow } from 'enzyme'

import { createStore } from 'redux'

import App from '../'

describe('App', () => {
  it('renders correctly', () => {
    const store = createStore(() => {})
    const history = {}
    const theme = {}
    const app = shallow(<App history={history} store={store} theme={theme} />)

    expect(app).toMatchSnapshot()
  })
})
