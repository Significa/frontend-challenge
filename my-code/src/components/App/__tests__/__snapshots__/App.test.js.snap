// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ConnectedRouter)
    history={Object {}}
  >
    <ThemeProvider
      theme={Object {}}
    >
      <Layout>
        <Switch>
          <Route
            component={[Function]}
            path="/movie"
          />
          <Route
            component={[Function]}
          />
        </Switch>
      </Layout>
    </ThemeProvider>
  </Connect(ConnectedRouter)>
</Provider>
`;
